apps/nextjs/src/stores/chatStore/actionFunctions/handleExecuteQueuedAction.ts

```diff
@@ -0,0 +1,31 @@
+export function handleExecuteQueuedAction(set, get) {
+  return async () => {
+    const { queuedUserAction, isExecutingAction, actions } = get();
+    const { append, reload } = actions;
+
+    if (!queuedUserAction || isExecutingAction) return;
```

Comment:
I might have mentioned this before, but I think we should rename this to `isExecutingQueuedAction`